flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ReadableCatalog.java

```diff
@@ -132,4 +134,58 @@
 	 */
 	boolean tableExists(ObjectPath objectPath) throws CatalogException;
 
+	// ------ partitions ------
+
+	/**
+	 * Gets PartitionSpec of all partitions of the table.
+	 *
+	 * @param tablePath		Path of the table.
+	 * @return	A list of PartitionSpec of the table.
+	 *
+	 * @throws TableNotExistException    thrown if the table does not exist in the catalog.
+	 * @throws TableNotPartitionedException		thrown if the table is not partitioned.
+	 * @throws CatalogException	in case of any runtime exception
+	 */
+	List<CatalogPartition.PartitionSpec> listPartitions(ObjectPath tablePath)
+		throws TableNotExistException, TableNotPartitionedException, CatalogException;
+
+	/**
+	 * Gets PartitionSpec of all partitions that is under the given PartitionSpec in the table .
+	 *
+	 * @param tablePath			Path of the table
+	 * @param partitionSpecs	The partition spec to list
+	 * @return A list of PartitionSpec that is under the given ParitionSpec in the table.
+	 *
+	 * @throws TableNotExistException    thrown if the table does not exist in the catalog.
+	 * @throws TableNotPartitionedException		thrown if the table is not partitioned.
+	 * @throws CatalogException	in case of any runtime exception
+	 */
+	List<CatalogPartition.PartitionSpec> listPartitions(ObjectPath tablePath, CatalogPartition.PartitionSpec partitionSpecs)
+		throws TableNotExistException, TableNotPartitionedException, CatalogException;
+
+	/**
+	 * Gets a partition of the given table.
+	 *
+	 * @param tablePath		Path of the table
+	 * @param partitionSpecs	Partition spec of partition to get
+	 * @return The requested partition.
+	 *
+	 * @throws TableNotExistException    thrown if the table does not exist in the catalog.
+	 * @throws TableNotPartitionedException		thrown if the table is not partitioned.
+	 * @throws PartitionNotExistException		thrown if the partition is not partitioned.
+	 * @throws CatalogException	in case of any runtime exception
+	 */
+	CatalogPartition getPartition(ObjectPath tablePath, CatalogPartition.PartitionSpec partitionSpecs)
```

Comment:
1. What happens if the given partitionSpec doesn't point to a single partition (, rather, a list of partitions)?
2. partitionSpecs -> partitionSpec?

Reply:
1. it will throw PartitionNotFoundException. I added additional javadoc for this behavior.
2. will rename it